fileUtils.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const lockMap = new Map();
+
+const writeFile = (path, src) => {
+    return new Promise(async (resolve, reject) => {
+        try {
+            if (lockMap.get(key) && !lockMap.get(key).locked) {
+                const fileHandle = await fs.promises.open(`${process.cwd()}/chunkSrc/${path}`, 'w+');
+                if (fileHandle) {
+                    await fs.promises.writeFile(fileHandle, src);
+                    resolve();
+                }
+            } else {
+                console.error('request for writeFile is touching a locked file, denied.')
```

Comment:
This must be allowed to continue once the previous client is done. We cannot throw.

Reply:
not used for locks here anymore